src/web3/KeychainAccountManager.ts

```diff
@@ -263,24 +229,31 @@ export class KeychainSigner implements Signer {
     return true
   }
 
-  async decrypt(ciphertext: Buffer): Promise<Buffer> {
-    return this.localSigner.decrypt(ciphertext)
-  }
-
-  async computeSharedSecret(publicKey: string): Promise<Buffer> {
-    return this.localSigner.computeSharedSecret(publicKey)
+  /**
+   * Get the unlocked contractkit signer. Throws if not unlocked.
+   */
+  get unlockedContractKitSigner() {
```

Comment:
OK, I [took a hacky pass at implementing option 3](https://github.com/valora-inc/wallet/compare/ethers-wallet...jophish/test-unlockable) using an `Unlockable` pseudo-mixin with decorators :sweat_smile:. I think I got a little nerd-sniped, but it's kind of neat. One benefit of having locks on the signers themselves is that we don't have to worry about keeping them private.

Reply:
Thanks, one thing we discussed with @jh2oman is that we needed the 2 signers unlocked at the same time, to handle the case where the private key needed to be imported again on the first unlock (migrating old users).

I don't have time right now to detail and check the new implementation can still do this, but I'll do tomorrow.

Reply:
@jh2oman and I had a (very!) long call on Friday where we came to a new approach for the entire refactor which makes things significantly simpler and a bit more elegant, involving moving the locking mechanism _down_ a layer of abstraction. I'm closing this PR and will open a new one with the new design, since the changes don't quite fit in this PR any longer.